### PR TITLE
[awsproxy] Expose service name as config option

### PR DESCRIPTION
**Pull request by arpitjindal97**
awsproxy extension can be used as proxy to any service not just xray

**Description:** <Describe what has changed.>
The service name variable was hardcoded to "xray". This PR exposes it to be configurable in the config.yaml

**Link to tracking Issue:** <Issue number if applicable>

**Testing:** <Describe what testing was performed and which tests were added.>

**Documentation:** <Describe the documentation added.>
Appropriate README of the extension has been updated to reflect the addition of new option

### CODE REVIEW

File: CHANGELOG.md
Comment (atoulme) [inline CHANGELOG.md:20]: please don't edit CHANGELOG.md, please run `make chlog-new` from the root of the repository and fill in the file created.

File: extension/awsproxy/extension.go
Comment (atoulme) [inline extension/awsproxy/extension.go:26]: can we make this rename a separate chore PR?

File: extension/awsproxy/factory.go
Comment (atoulme) [inline extension/awsproxy/factory.go:46]: as above

File: internal/aws/proxy/server.go
Comment (atoulme) [inline internal/aws/proxy/server.go:103]: any consequence if `ServiceName` is empty or consists of an invalid service such as "foo"?

File: CHANGELOG.md
Comment (arpitjindal97) [reply CHANGELOG.md]: Fixed

File: extension/awsproxy/extension.go
Comment (arpitjindal97) [reply extension/awsproxy/extension.go:26]: Sure, I was thinking about this too

File: internal/aws/proxy/server.go
Comment (arpitjindal97) [reply internal/aws/proxy/server.go:103]: so, I just ran the new code leaving `service_name` as empty quotes and OTEL starts just fine:
```
$ curl http://localhost:2000/                                                                                                     
{"message":"Credential should be scoped to correct service: 'es'. "}
```

Comment (arpitjindal97) [reply internal/aws/proxy/server.go:103]: I'm testing it with an OpenSearch endpoint. Hence, my AWS is telling me to put "es" as service name which is expected.

Comment (arpitjindal97) [reply internal/aws/proxy/server.go:103]: I also tried with `foo` and get same response. So, I can safely say, there are no consequences

Comment (atoulme) [reply internal/aws/proxy/server.go:103]: Well, sure. It fails gracefully with an error message that makes sense, so there's that.

File: .chloggen/main.yaml
Comment (atoulme) [inline .chloggen/main.yaml:13]: Please enter an issue number; use this PR number if no issue is filed.

File: internal/aws/proxy/cfg.go
Comment (atoulme) [inline internal/aws/proxy/cfg.go:41]: Add some godoc to the field please

Comment (arpitjindal97) [reply internal/aws/proxy/cfg.go:41]: Sorry, missed by mistake. I have added now

File: .chloggen/main.yaml
Comment (arpitjindal97) [reply .chloggen/main.yaml]: Done

**Review by atoulme — APPROVED** [review]
LGTM.

**Comment by arpitjindal97** [comment]
@atoulme Can you approve the pipeline?

**Comment by arpitjindal97** [comment]
@atoulme Can you approve the pipeline once again? some test were failing. I have fixed them.

**Comment by arpitjindal97** [comment]
@atoulme All test passed. Can we merge it?

**Comment by atoulme** [comment]
We need a review from codeowners. @dmitryax @Aneurysm9 please review

**Comment by arpitjindal97** [comment]
@bryan-aguilar Can you help in getting approval from your colleague **Anthony Mirabella**?

**Comment by arpitjindal97** [comment]
@Aneurysm9 Gentle reminder

**Comment by arpitjindal97** [comment]
@Aneurysm9 Gentle reminder

**Comment by arpitjindal97** [comment]
@Aneurysm9 Gentle reminder

**Comment by arpitjindal97** [comment]
@Aneurysm9 Gentle reminder

**Comment by arpitjindal97** [comment]
Hello team,

It has been 3 months since this PR is open. Seems like @Aneurysm9 is not active on this community anymore.

how do you plan to continue the development of this plugin moving forward?

cc: @atoulme @mxiamxia @djaglowski

**Review by bryan-aguilar — APPROVED** [review]

**Review by mxiamxia — APPROVED** [review]
LGTM

**Comment by bryan-aguilar** [comment]
```
cd /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/tools && go build -o /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/builder -trimpath go.opentelemetry.io/collector/cmd/builder
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/builder --skip-compilation --config cmd/otelcontribcol/builder-config.yaml --output-path cmd/otelcontribcol
Flag --output-path has been deprecated, use config distribution::output_path
2024-02-27T20:54:39.623Z	INFO	internal/command.go:123	OpenTelemetry Collector Builder	{"version": "", "date": "unknown"}
2024-02-27T20:54:39.628Z	INFO	internal/command.go:159	Using config file	{"path": "cmd/otelcontribcol/builder-config.yaml"}
2024-02-27T20:54:39.628Z	INFO	builder/config.go:109	Using go	{"go-executable": "/opt/hostedtoolcache/go/1.21.7/x64/bin/go"}
2024-02-27T20:54:39.628Z	INFO	builder/main.go:67	You're building a distribution with non-aligned version of the builder. Compilation may fail due to API changes. Please upgrade your builder or API	{"builder-version": "0.94.0"}
2024-02-27T20:54:39.636Z	INFO	builder/main.go:91	Sources created	{"path": "cmd/otelcontribcol"}
2024-02-27T20:54:39.636Z	INFO	builder/main.go:25	Running go subcommand.	{"arguments": ["get", "cloud.google.com/go"]}
2024-02-27T20:54:41.684Z	INFO	builder/main.go:25	Running go subcommand.	{"arguments": ["mod", "tidy", "-compat=1.21"]}
2024-02-27T20:54:43.726Z	INFO	builder/main.go:142	Getting go modules
2024-02-27T20:54:43.726Z	INFO	builder/main.go:25	Running go subcommand.	{"arguments": ["mod", "download"]}
2024-02-27T20:54:44.331Z	INFO	builder/main.go:98	Generating source codes only, the distribution will not be compiled.
make --no-print-directory -C cmd/otelcontribcol fmt
Makefile:4: warning: overriding recipe for target 'lint'
../../Makefile.Common:199: warning: ignoring old recipe for target 'lint'
cd /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/internal/tools && go build -o /home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/goimports -trimpath golang.org/x/tools/cmd/goimports
gofmt  -w -s ./
/home/runner/work/opentelemetry-collector-contrib/opentelemetry-collector-contrib/.tools/goimports -w  -local github.com/open-telemetry/opentelemetry-collector-contrib ./
Generated code is out of date, please run "make genotelcontribcol" and commit the changes in this PR.
```

**Comment by arpitjindal97** [comment]
Hello @bryan-aguilar ,

I have rebased my branch just now with latest main. Can you run the pipeline again?

**Comment by arpitjindal97** [comment]
Seems like issues with podman permission, not related to my changes

```
# test install
echo
podman run --name $container_name -d $image_name
Error: OCI runtime error: chmod `run/shm`: Operation not supported
podman rm -fv otelcontribcol-deb-test >/dev/null 2>&1 || true
```

**Comment by bryan-aguilar** [comment]
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/31443

**Comment by arpitjindal97** [comment]
@bryan-aguilar I have rebased my branch again. Can you please run the CI again.

For some reason Github is showing 2 commits, but you can checkout locally it's just 1 commit.

**Comment by arpitjindal97** [comment]
@bryan-aguilar nevermind, github is showing correct commit now. It’s ready to merge ✌️

**Review by codeboten — APPROVED** [review]

**Pull request merged**
